shared/driver/conn.go

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package driver
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+)
+
+// Conn is a DB driver conn implementation
+// which will just pass-through all queries to its
+// underlying connection.
+type Conn struct {
+	conn *sql.Conn
+}
+
+// driverConn is a super-interface combining the basic
+// driver.Conn interface with some new additions later.
+type driverConn interface {
+	driver.Conn
+	driver.ConnBeginTx
+	driver.ConnPrepareContext
+	driver.Execer
+	driver.ExecerContext
+	driver.Queryer
+	driver.QueryerContext
+	driver.Pinger
+}
+
+var (
+	// Compile-time check to ensure Conn implements the interface.
+	_ driverConn = &Conn{}
+)
+
+func (c *Conn) Begin() (tx driver.Tx, err error) {
+	err = c.conn.Raw(func(innerConn interface{}) error {
+		tx, err = innerConn.(driver.Conn).Begin()
```

Comment:
Should I ", ok" check everywhere? The pg and mysql drivers we use already satisfy
all the interfaces and this will change when we eventually move to RPC. But let me
know if you feel it's needed. 0/5.